### ISSUE TITLE
 cucumber failures with puppet 4.0.0 / hiera 2.0.0

### ISSUE THREAD

**Issue by jhoblitt** [TomPoulton/hiera-eyaml#155]
Multiple acceptance tests are failing with the Gemfile from the current master which allows puppet 4.0.0 / hiera 2.0.0.

```
[master] ~/github/hiera-eyaml $ git show-ref HEAD
54b62eff28d72cb8670df226e2a16b7562aa16fc refs/remotes/origin/HEAD
[master] ~/github/hiera-eyaml $ ruby --version
ruby 2.1.5p273 (2014-11-13 revision 48405) [x86_64-linux]
[master] ~/github/hiera-eyaml $ git diff
[master] ~/github/hiera-eyaml $ bundle show puppet
/home/jhoblitt/github/hiera-eyaml/.bundle/gems/puppet-4.0.0
[master] ~/github/hiera-eyaml $ bundle show hiera
/home/jhoblitt/github/hiera-eyaml/.bundle/gems/hiera-2.0.0
[master] ~/github/hiera-eyaml $ bundle show cucumber
/home/jhoblitt/github/hiera-eyaml/.bundle/gems/cucumber-1.3.19
```
```
  Scenario: encrypt a password                                        # features/encrypts.feature:23
    When I run `./supply_password.sh eyaml encrypt -o string -p`      # .bundle/gems/aruba-0.6.2/lib/aruba/cucumber.rb:93
      No such file or directory - ./supply_password.sh (Aruba::LaunchError)
      features/encrypts.feature:24:in `When I run `./supply_password.sh eyaml encrypt -o string -p`'
    Then the file "password.output" should match /ENC\[PKCS7,(.*?)\]/ # .bundle/gems/aruba-0.6.2/lib/aruba/cucumber.rb:351
```
```
  Scenario: verify puppet3 with hiera can use hiera-eyaml to decrypt data                             # features/puppet.feature:7
    When I run `rm -f /tmp/eyaml_puppettest.* 2>/dev/null`                                            # .bundle/gems/aruba-0.6.2/lib/aruba/cucumber.rb:93
    When I run `puppet apply --confdir ./puppet --node_name_value localhost puppet/manifests/init.pp` # .bundle/gems/aruba-0.6.2/lib/aruba/cucumber.rb:93
    Then the file "/tmp/eyaml_puppettest.1" should match /^good night$/                               # .bundle/gems/aruba-0.6.2/lib/aruba/cucumber.rb:351
      expected "rise and shine\n" to match /^good night$/
      Diff:
      @@ -1,2 +1,2 @@
      -/^good night$/
      +rise and shine
       (RSpec::Expectations::ExpectationNotMetError)
      features/puppet.feature:10:in `Then the file "/tmp/eyaml_puppettest.1" should match /^good night$/'
    Then the file "/tmp/eyaml_puppettest.2" should match /^and good luck$/                            # .bundle/gems/aruba-0.6.2/lib/aruba/cucumber.rb:351
    Then the file "/tmp/eyaml_puppettest.3" should match /^and good luck$/                            # .bundle/gems/aruba-0.6.2/lib/aruba/cucumber.rb:351
    Then the file "/tmp/eyaml_puppettest.4" should match /^and good luck$/                            # .bundle/gems/aruba-0.6.2/lib/aruba/cucumber.rb:351
    Then the file "/tmp/eyaml_puppettest.5" should match /^gangs of new york$/                        # .bundle/gems/aruba-0.6.2/lib/aruba/cucumber.rb:351
```
```
  Scenario: verify puppet3 and facter for correct hash merge with incorrect fact                                              # features/puppet.feature:17
    Given I set FACTER_fact to "not-existcity"                                                                                # features/support/puppet.rb:1
    When I run `rm -f /tmp/eyaml_puppettest.* 2>/dev/null`                                                                    # .bundle/gems/aruba-0.6.2/lib/aruba/cucumber.rb:93
    When I run `puppet apply --confdir ./puppet-hiera-merge --node_name_value localhost puppet-hiera-merge/manifests/init.pp` # .bundle/gems/aruba-0.6.2/lib/aruba/cucumber.rb:93
    Then the file "/tmp/eyaml_puppettest.1" should match /^good night$/                                                       # .bundle/gems/aruba-0.6.2/lib/aruba/cucumber.rb:351
      expected "rise and shine\n" to match /^good night$/
      Diff:
      @@ -1,2 +1,2 @@
      -/^good night$/
      +rise and shine
       (RSpec::Expectations::ExpectationNotMetError)
      features/puppet.feature:22:in `Then the file "/tmp/eyaml_puppettest.1" should match /^good night$/'
    Then the file "/tmp/eyaml_puppettest.2" should match /^great to see you$/                                                 # .bundle/gems/aruba-0.6.2/lib/aruba/cucumber.rb:351
    Then the file "/tmp/eyaml_puppettest.3" should match /good luck/                                                          # .bundle/gems/aruba-0.6.2/lib/aruba/cucumber.rb:351
    Then the file "/tmp/eyaml_puppettest.4" should match /"here": "we go again!"/                                             # .bundle/gems/aruba-0.6.2/lib/aruba/cucumber.rb:351
    Then the file "/tmp/eyaml_puppettest.5" should match /^gangs of new york\nis to the warriors$/                            # .bundle/gems/aruba-0.6.2/lib/aruba/cucumber.rb:351
```

**Comment by TomPoulton** [comment]
Hi @jhoblitt, sorry for the (very) late reply!

There's some interesting stuff here, do you think this is a hiera-eyaml or a cucumber (aruba) incompatibility with puppet 4.0? The load error seems like it's a potentially normal issue, but eyaml returning the wrong decrypted values seems very strange!

**Comment by sihil** [comment]
Closing this as solved by #181.
